Dockerfile

```diff
@@ -18,18 +18,19 @@ ENV GOOS=linux
 ENV CGO_ENABLED=0
 ENV VERSION_PKG=github.com/aws-controllers-k8s/$service_alias-controller/pkg/version
 # Copy the Go Modules manifests and LICENSE/ATTRIBUTION
-COPY LICENSE $work_dir/LICENSE
-COPY ATTRIBUTION.md $work_dir/ATTRIBUTION.md
-COPY go.mod $work_dir/go.mod
-COPY go.sum $work_dir/go.sum
+COPY $service_alias-controller/LICENSE $work_dir/LICENSE
+COPY $service_alias-controller/ATTRIBUTION.md $work_dir/ATTRIBUTION.md
+# Copy Go mod files
+COPY $service_alias-controller/go.mod $work_dir/go.mod
+COPY $service_alias-controller/go.sum $work_dir/go.sum
```

Comment:
This `Dockerfile` was used with build context: `SERVICE_CONTROLLER_SOURCE_PATH` thus statements like `COPY go.sum $work_dir/go.sum` worked fine.

However as in this PR, DOCKER_BUILD_CONTEXT is used for this Dockerfile; updates like `COPY $service_alias-controller/go.sum $work_dir/go.sum` are required.

I am fine with this change, as this brings Dockerfile, and Dockerfile.local to have similar contents interms of resolving paths.